include/boost/gil/extension/numeric/algorithm.hpp

```diff
@@ -248,6 +250,141 @@ void view_multiplies_scalar(SrcView const& src_view, Scalar const& scalar, DstVi
     }
 }
 
+
+/// \ingroup ImageAlgorithms
+/// \brief Boundary options for image boundary extension
+enum class boundary_option
+{
+    boundary_option_extend_zero,    /// assume the source boundaries to be zero
```

Comment:
There is no need to keep the "boundary_option" prefix in the enumerator name, as the (enum) class already provides the named scope, i.e. users can just write "boundary_option::extend_zero" rather than "boundary_option::boundary_option_extend_zero".

Reply:
Good point

Reply:
That made complete sense. I have updated the code.